docs/scripts.md

```diff
@@ -0,0 +1,635 @@
+node file.js
+```
+
+### Global variables
+
+```js
+// zx
+await $`echo example`;
+```
+
+```js
+// Execa
+import {$} from 'execa';
+await $`echo example`;
+```
+
+### Command execution
+
+```sh
+# Bash
+echo example
+```
+
+```js
+// zx
+await $`echo example`;
+```
+
+```js
+// Execa
+await $`echo example`;
+```
+
+### Subcommands
+
+```sh
+# Bash
+echo "$(echo example)"
+```
+
+```js
+// zx
+const example = await $`echo example`;
+await $`echo ${example}`;
+```
+
+```js
+// Execa
+const example = await $`echo example`;
+await $`echo ${example}`;
+```
+
+### Parallel commands
+
+```sh
+# Bash
+echo one &
+echo two &
+```
+
+```js
+// zx
+await Promise.all([$`echo one`, $`echo two`]);
+```
+
+```js
+// Execa
+await Promise.all([$`echo one`, $`echo two`]);
+```
+
+### Serial commands
+
+```sh
+# Bash
+echo one && echo two
+```
+
+```js
+// zx
+await $`echo one && echo two`;
+```
+
+```js
+// Execa
+await $`echo one`;
+await $`echo two`;
+```
+
+### Local binaries
+
+```sh
+# Bash
+npx tsc --version
+```
+
+```js
+// zx
+await $`npx tsc --version`;
+```
+
+```js
+// Execa
+await $`tsc --version`;
+```
+
+### Builtin utilities
+
+```js
+// zx
+const content = await stdin();
+```
+
+```js
+// Execa
+import getStdin from 'get-stdin';
+const content = await getStdin();
+```
+
+### Variable substitution
+
+```sh
+# Bash
+echo $LANG
+```
+
+```js
+// zx
+await $`echo $LANG`;
+```
+
+```js
+// Execa
+await $`echo ${process.env.LANG}`;
+```
+
+### Environment variables
+
+```sh
+# Bash
+EXAMPLE=1 example_command
+```
+
+```js
+// zx
+$.env.EXAMPLE = '1';
+await $`example_command`;
+delete $.env.EXAMPLE;
+```
+
+```js
+// Execa
+await $({env: {EXAMPLE: '1'}})`example_command`;
+```
+
+### Escaping
+
+```sh
+# Bash
+echo 'one two'
+```
+
+```js
+// zx
+await $`echo ${'one two'}`;
+```
+
+```js
+// Execa
+await $`echo ${'one two'}`;
+```
+
+### Escaping multiple arguments
+
+```sh
+# Bash
+echo 'one two' '$'
+```
+
+```js
+// zx
+await $`echo ${['one two', '$']}`;
+```
+
+```js
+// Execa
+await $`echo ${['one two', '$']}`;
+```
+
+### Current filename
+
+```sh
+# Bash
+echo "$(basename "$0")"
+```
+
+```js
+// zx
+await $`echo ${__filename}`;
+```
+
+```js
+// Execa
+import {fileURLToPath} from "node:url";
+import {basename} from "node:path";
+const __filename = basename(fileURLToPath(import.meta.url));
+
+await $`echo ${__filename}`;
+```
+
+### Verbose mode
+
+```sh
+# Bash
+set -v
+echo example
+```
+
+```js
+// zx >=8
+await $`echo example`.verbose();
+
+// or:
+$.verbose = true;
+```
+
+```js
+// Execa
+const $$ = $({verbose: true});
+await $$`echo example`;
+```
+
+Or:
+
+```
+NODE_DEBUG=execa node file.js
+```
+
+### Current directory
+
+```sh
+# Bash
+cd project
+```
+
+```js
+// zx
+cd('project');
+
+// or:
+$.cwd = 'project';
+```
+
+```js
+// Execa
+const $$ = $({cwd: 'project'});
+```
+
+### Multiple current directories
+
+```sh
+# Bash
+pushd project
+pwd
+popd
+pwd
+```
+
+```js
+// zx
+within(async () => {
```

Comment:
This does not seem to be the case according to their documentation: https://github.com/google/zx/#within
Although according to their type, it would: https://github.com/google/zx/blob/4999435182fe1f34f6b2ef2471184d9878456b5e/src/core.ts#L436 :thinking:

Reply:
Wonder if that's a typo. I saw that and their second example including an await/assignment.  Not a show stopper imo